app/src/main/java/com/github/sdp_begreen/begreen/activities/SharePostActivity.kt

```diff
@@ -0,0 +1,111 @@
+package com.github.sdp_begreen.begreen.activities
+
+import android.graphics.Bitmap
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.EditText
+import android.widget.ImageView
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.models.Post
+
+class SharePostActivity : AppCompatActivity() {
+
+    private lateinit var postImageView : ImageView;
+    private lateinit var postTitleEditText : EditText;
+    private lateinit var sharePostBtn : Button;
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_share_post)
+
+        setupUI()
+    }
+
+    /**
+     * Helper function to setups ui buttons
+     */
+    private fun setupUI() {
+
+        // Bind all UIs
+        postTitleEditText = findViewById(R.id.postTitleEditText);
+        sharePostBtn = findViewById(R.id.sharePostBtn)
+        postImageView = findViewById(R.id.postImageView);
+
+        // Display the post image.
+        displayPostImage()
+
+        // Setup the share button action
+        sharePostBtn.setOnClickListener {
+            sharePost()
+        }
+    }
+
+    /**
+     * Helper function to return the image post which is an extra of the intent
+     */
+    private fun getPostImage() : Bitmap? {
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
```

Comment:
This is the 2nd time that code appears in as many files, perhaps it's worth extracting that logic into a "compatibility" utility somewhere